backend/src/src-cliservice/src/main/java/com/siemens/sw360/cliservice/CLIHandler.java

```diff
@@ -0,0 +1,121 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package com.siemens.sw360.cliservice;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Sets;
+import com.siemens.sw360.attachments.db.AttachmentDatabaseHandler;
+import com.siemens.sw360.cliservice.parsers.CLIParser;
+import com.siemens.sw360.cliservice.parsers.CopyrightLicenseInfoParser;
+import com.siemens.sw360.datahandler.common.DatabaseSettings;
+import com.siemens.sw360.datahandler.couchdb.AttachmentConnector;
+import com.siemens.sw360.datahandler.thrift.attachments.Attachment;
+import com.siemens.sw360.datahandler.thrift.attachments.AttachmentContent;
+import com.siemens.sw360.datahandler.thrift.attachments.AttachmentType;
+import com.siemens.sw360.datahandler.thrift.cliservice.CopyrightLicenseInfo;
+import com.siemens.sw360.datahandler.thrift.cliservice.CopyrightLicenseInfoService;
+import com.siemens.sw360.datahandler.thrift.components.Release;
+import org.apache.thrift.TException;
+
+import java.net.MalformedURLException;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.siemens.sw360.datahandler.common.CommonUtils.nullToEmptyList;
+import static com.siemens.sw360.datahandler.common.CommonUtils.nullToEmptySet;
+
+/**
+ * Implementation of the Thrift service
+ *
+ * @author alex.borodin@evosoft.com
+ */
+public class CLIHandler implements CopyrightLicenseInfoService.Iface {
+
+    private final AttachmentDatabaseHandler attachmentDatabaseHandler;
+
+    private final CopyrightLicenseInfoParser[] parsers;
+
+    public CLIHandler() throws MalformedURLException {
+        this(new AttachmentDatabaseHandler(DatabaseSettings.COUCH_DB_URL, DatabaseSettings.COUCH_DB_DATABASE, DatabaseSettings.COUCH_DB_ATTACHMENTS));
+
+    }
+
+    @VisibleForTesting
+    public CLIHandler(AttachmentDatabaseHandler handler) throws MalformedURLException {
+        attachmentDatabaseHandler = handler;
+        parsers = new CopyrightLicenseInfoParser[]{
+//                new SPDXParser(attachmentDatabaseHandler.getAttachmentConnector()),
+                new CLIParser(attachmentDatabaseHandler.getAttachmentConnector()),};
+    }
+
+    @Override
+    public CopyrightLicenseInfo getComponentLicenseInfoForAttachment(String attachmentContentId) throws TException {
+        AttachmentContent attachmentContent = attachmentDatabaseHandler.getAttachmentContent(attachmentContentId);
+        return applyParsers(attachmentContent);
+    }
+
+    @Override
+    public CopyrightLicenseInfo getComponentLicenseInfoForRelease(Release release) throws TException {
+        try {
+            CopyrightLicenseInfo result = nullToEmptySet(release.getAttachments()).stream()
+                    .filter(a -> a.getAttachmentType() == AttachmentType.COMPONENT_LICENSE_INFO_XML)
```

Comment:
No. This is a draft.

Yes.

Yes, I think there was.

I had this idea of separating filtering which attachments are worth considering at all from actually checking whether they can be parsed. Now I see another problem with it. I'll rewrite this.